sympy/utilities/codegen.py

```diff
@@ -695,6 +695,15 @@ def routine(self, name, expr, argument_sequence=None, global_vars=None):
         arg_list = []
 
         # setup input argument list
+
+        # helper to add dimension data for array args
+        def dimensions(s):
+            dims = []
```

Comment:
or just `[(S.Zero, dim - 1) for dim in s.shape]`

Reply:
Much better, yes. Done!